pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -129,12 +129,52 @@ class Dot(Module):
   is_original_torch_module = False
 
   def create_returnn_layer_dict(self, a: Tensor, b: Tensor, *, reduce_dim_a: int, reduce_dim_b: int) -> Dict[str, Any]:
-    assert a.shape == b.shape
     dim0 = self._get_input_axis_to_returnn(a, reduce_dim_a)
     dim1 = self._get_input_axis_to_returnn(a, reduce_dim_b)
     sources = [self._get_input_layer_name(source) for source in [a, b]]
     return {"class": "dot", "red1": dim0, "red2": dim1, "from": sources}
 
+  def _get_output_shape_from_returnn(self, inputs_flat: List[Tensor], layer: LayerBase
```

Comment:
The order of entries in the torch output shape is wrong. But maybe this is actually the same issue as above.

Reply:
When I remove this, I end up in `assert len(rem_out) == len(rem_in)  # assumption, otherwise no idea what to do`. The `DotLayer` moves unmatched axes to the end and this is not handled in the general case.